---
title: "11_1_module_GO-MWU_prep_WGCNA_clusters"
author: "Aidan Coyle"
date: "Last compiled on `r format(Sys.time(), '%Y-%m-%d')`"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

Previously, we ran GO-MWU on a variety of pairwise comparisons for each transcriptome. That's useful, but there's another way to use GO-MWU.

Essentially, you take a module that was deemed significant by WGCNA, and analyze that in the context of all genes that were analyzed by WGCNA.

To do this, you give GO-MWU two inputs.

- A 2-column table of accession IDs and kME values (with the kME values of genes outside the module set to 0). A header line is required, although the contents are irrelevant.

- A tab-delimited 2-column table of accession IDs and GO terms with no header and only 1 line per gene.

This script produces the CSV needed for GO-MWU, which is a 2-column table of accession IDs and log2 fold changes. A header line is required, although the contents are irrelevant.

GO-MWU also requires another input file which we already created - a tab-delimited 2-column table of accession IDs and GO terms with no header and only 1 line per gene

In this script, we'll work to create the first of those two files.

First, here's a list of all significant modules for all 3 transcriptomes where the association isn't due to a single crab. Names correspond to what they're called in the "day" analysis for that respective transcriptome/comparison combination

| Transcriptome | Module | Trait(s)                               | p-value(s)    |
|---------------|--------|----------------------------------------|---------------|
| Complete      | yellow | Lowered vs. Elevated, Elevated vs. All | 0.02, 0.03    |
| Complete      | cyan   | Lowered vs. Ambient, Elevated vs. All  | 0.02, 0.04    |
| Complete      | salmon | Lowered vs. Ambient                    | 0.04          |
| Complete      | blue   | day                                    | 0.03          |
| Tanner crab   | black  | Lowered vs. Ambient, Elevated vs. All  | 8x10^-4, 0.05 |
| Tanner crab   | brown  | Lowered vs. Elevated, Elevated vs. All | 0.02, 0.03    |
| Parasite      | brown  | Hematodinium level                     | 0.02          |
| Parasite      | pink   | day                                    | 0.03          |

We also ran WGCNA on only Lowered and Ambient-temperature crab to examine the effect of changes in Hematodinium level. Here's significant modules (again, where association is not due to expression in a single crab). For all, Trait is hemat_diff (change in Hematodinium level on log scale), as other significant traits were discarded

| Transcriptome | Module | p-value |
|---------------|--------|---------|
| Complete      | blue   | 0.02    |
| Tanner crab   | red    | 0.04    |
| Parasite      | green  | 0.05    |

Alright, let's get started:

First, load all libraries as usual

```{r libraries, message = FALSE, warning=FALSE}
# Add all required libraries that are installed with install.packages() here
list.of.packages <- "tidyverse"
# Add all libraries that are installed using BiocManager here
bioconductor.packages <- ""

# Install BiocManager if needed
if(!requireNamespace("BiocManager", quietly = TRUE)) install.packages("BiocManager")

# Get names of all required packages that aren't installed
new.packages <- list.of.packages[!(list.of.packages %in% installed.packages()[, "Package"])]
new.bioc.packages <- bioconductor.packages[!(bioconductor.packages %in% installed.packages()[, "Package"])]
# Install all new packages
if(length(new.packages)) install.packages(new.packages)
if(length(new.bioc.packages)) BiocManager::install(new.bioc.packages)

# Load all required libraries
all.packages <- c(list.of.packages, bioconductor.packages)
lapply(all.packages, FUN = function(X) {
  do.call("require", list(X))
})

source("hematodinium_analysis_functions.R")

```


We'll now repeatedly run WGCNA_modules_accessions_KMEs()

This works as follows:


Inputs:
- The transcripts within a module produced by WGCNA
- All kMEs for all modules in your WGCNA run (should be in same directory as significant module)
- A BLAST file for your relevant transcriptome


Output:
  - A tab-separated two-column table of accession IDs and module kME values for all genes (kME values for genes not in module should be set to 0)
      - This goes in the output_GOMWU directory
  - A single column of accession IDs (this will be used to create the second input for GO-MWU)
      - This will be put in the output_accessions directory
      
Function arguments:
- module_name: color of the module you want to examine
- transcriptome: name of the relevant transcriptome
- compar: libraries used in WGCNA. Should be all_crabs or amb_low
- key_var: variable selected as main variable to examine in WGCNA run
- blast_path: path to the BLAST file for your relevant transcriptome
- output_GOMWU: path to the folder you want to put your files in (should be the one you run GO-MWU in)
- output_accessions: path to the folder you want to put your single column of accession IDs in

## ALL CRABS

This section analyzes output from WGCNA analysis of all crabs

```{r inputs}

# COMPLETE TRANSCRIPTOME ------------------------------

# Yellow module, cbai2.0, all crabs
WGCNA_modules_accessions_kMEs(module_name = "yellow",
                              transcriptome = "cbai_transcriptomev2.0",
                              compar = "all_crabs",
                              key_var = "day",
                              blast_path = "../data/blast_tables/cbai_transcriptomev2.0_blast_table.txt",
                              output_GOMWU = "../scripts/11_4_running_GOMWU/",
                              output_accessions = "../output/accession_n_GOids/WGCNA_modules/")

# Cyan module, cbai2.0, all crabs
WGCNA_modules_accessions_kMEs(module_name = "cyan",
                              transcriptome = "cbai_transcriptomev2.0",
                              compar = "all_crabs",
                              key_var = "day",
                              blast_path = "../data/blast_tables/cbai_transcriptomev2.0_blast_table.txt",
                              output_GOMWU = "../scripts/11_4_running_GOMWU/",
                              output_accessions = "../output/accession_n_GOids/WGCNA_modules/")

# Salmon module, cbai2.0, all crabs
WGCNA_modules_accessions_kMEs(module_name = "salmon",
                              transcriptome = "cbai_transcriptomev2.0",
                              compar = "all_crabs",
                              key_var = "day",
                              blast_path = "../data/blast_tables/cbai_transcriptomev2.0_blast_table.txt",
                              output_GOMWU = "../scripts/11_4_running_GOMWU/",
                              output_accessions = "../output/accession_n_GOids/WGCNA_modules/")

# Blue module, cbai2.0, all crabs
WGCNA_modules_accessions_kMEs(module_name = "blue",
                              transcriptome = "cbai_transcriptomev2.0",
                              compar = "all_crabs",
                              key_var = "day",
                              blast_path = "../data/blast_tables/cbai_transcriptomev2.0_blast_table.txt",
                              output_GOMWU = "../scripts/11_4_running_GOMWU/",
                              output_accessions = "../output/accession_n_GOids/WGCNA_modules/")

#### TANNER CRAB TRANSCRIPTOME -------------------------------

# Black module, cbai4.0, all crabs
WGCNA_modules_accessions_kMEs(module_name = "black",
                              transcriptome = "cbai_transcriptomev4.0",
                              compar = "all_crabs",
                              key_var = "day",
                              blast_path = "../data/blast_tables/cbai_transcriptomev4.0_blast_table.txt",
                              output_GOMWU = "../scripts/11_4_running_GOMWU/",
                              output_accessions = "../output/accession_n_GOids/WGCNA_modules/")

# Brown module, cbai4.0, all crabs
WGCNA_modules_accessions_kMEs(module_name = "brown",
                              transcriptome = "cbai_transcriptomev4.0",
                              compar = "all_crabs",
                              key_var = "day",
                              blast_path = "../data/blast_tables/cbai_transcriptomev4.0_blast_table.txt",
                              output_GOMWU = "../scripts/11_4_running_GOMWU/",
                              output_accessions = "../output/accession_n_GOids/WGCNA_modules/")


#### PARASITE TRANSCRIPTOME -------------------------

# Brown module, hemat1.6, all crabs
WGCNA_modules_accessions_kMEs(module_name = "brown",
                              transcriptome = "hemat_transcriptomev1.6",
                              compar = "all_crabs",
                              key_var = "day",
                              blast_path = "../data/blast_tables/hemat_transcriptomev1.6_blast_table.txt",
                              output_GOMWU = "../scripts/11_4_running_GOMWU/",
                              output_accessions = "../output/accession_n_GOids/WGCNA_modules/")

# Pink module, hemat1.6, all crabs
WGCNA_modules_accessions_kMEs(module_name = "pink",
                              transcriptome = "hemat_transcriptomev1.6",
                              compar = "all_crabs",
                              key_var = "day",
                              blast_path = "../data/blast_tables/hemat_transcriptomev1.6_blast_table.txt",
                              output_GOMWU = "../scripts/11_4_running_GOMWU/",
                              output_accessions = "../output/accession_n_GOids/WGCNA_modules/")


```



## Ambient and Low Crabs

This section analyzes output from WGCNA analysis of only Ambient and Low-temperature samples.

It specifically only looks at modules that are associated with change in _Hematodinium_ expression over the course of the experiment.

```{r inputs}

# COMPLETE TRANSCRIPTOME ------------------------------

# Blue module, cbai2.0, ambient and low
WGCNA_modules_accessions_kMEs(module_name = "blue",
                              transcriptome = "cbai_transcriptomev2.0",
                              compar = "amb_low",
                              key_var = "hemat_diff",
                              blast_path = "../data/blast_tables/cbai_transcriptomev2.0_blast_table.txt",
                              output_GOMWU = "../scripts/11_4_running_GOMWU/",
                              output_accessions = "../output/accession_n_GOids/WGCNA_modules/")

#### TANNER CRAB TRANSCRIPTOME -------------------------------

# Red module, cbai4.0, ambient and low
WGCNA_modules_accessions_kMEs(module_name = "red",
                              transcriptome = "cbai_transcriptomev4.0",
                              compar = "amb_low",
                              key_var = "hemat_diff",
                              blast_path = "../data/blast_tables/cbai_transcriptomev4.0_blast_table.txt",
                              output_GOMWU = "../scripts/11_4_running_GOMWU/",
                              output_accessions = "../output/accession_n_GOids/WGCNA_modules/")


#### PARASITE TRANSCRIPTOME -------------------------

# Green module, hemat1.6, all crabs
WGCNA_modules_accessions_kMEs(module_name = "green",
                              transcriptome = "hemat_transcriptomev1.6",
                              compar = "amb_low",
                              key_var = "hemat_diff",
                              blast_path = "../data/blast_tables/hemat_transcriptomev1.6_blast_table.txt",
                              output_GOMWU = "../scripts/11_4_running_GOMWU/",
                              output_accessions = "../output/accession_n_GOids/WGCNA_modules/")
```

