---
title: "12_1_supp_file_creation"
author: "Aidan Coyle"
date: "12/6/2021"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Supplemental File Creation

In this script, I will create the supplemental files for the paper.

Since the numbering hasn't been nailed down yet, I'll name the sections by describing the table I'm creating, rather than with numbering.

## Load libraries

```{r}
# Add all required libraries that are installed with install.packages() here
list.of.packages <- "tidyverse"

# Get names of all required packages that aren't installed
new.packages <- list.of.packages[!(list.of.packages %in% installed.packages()[, "Package"])]
# Install all new packages
if(length(new.packages)) install.packages(new.packages)

# Load all required libraries
all.packages <- list.of.packages
lapply(all.packages, FUN = function(X) {
  do.call("require", list(X))
})
```

## Annotation Tables

I'll create an annotation file for each transcriptome. This will be done by merging the existing annotation table with the GO terms for that annotation.

Annotated tables and the GO term database have been downloaded in previous scripts, but for the sake of reproducibility, here are the links to each:

**Transcriptome Annotation Tables**

All are in outfmt 6 and were annotated with BLASTx

[Complete transcriptome, AKA cbai_transcriptome_v2.0](https://gannet.fish.washington.edu/Atumefaciens/20200508_cbai_diamond_blastx_transcriptome-v2.0/20200507.C_bairdi.Trinity.blastx.outfmt6). MD5 is ace82a75cb947574ac807d868427253c

[Tanner crab transcriptome, AKA cbai_transcriptome_v4.0](https://gannet.fish.washington.edu/Atumefaciens/20210318_cbai_diamond_blastx_transcriptome-v4.0/cbai_transcriptome_v4.0.blastx.outfmt6). MD5 is 8fd2ab9c27e59653fcb4f32574be1f61

[Hematodinium transcriptome, AKA hemat_transcriptomev1.6](https://gannet.fish.washington.edu/Atumefaciens/20200814_hemat_diamond_blastx_v1.6_v1.7_v2.1_v3.1/hemat_transcriptome_v1.6.fasta.blastx.outfmt6). MD5 is 698513d787f92a65debb9f70dbb2e32f

**GO terms**

To get GO terms, we downloaded the full Swiss-ProtKB/Uniprot database (which includes all GO terms) on 2021-02-09 from https://www.uniprot.org/uniprot/
MD5sum for that download was ace82a75cb947574ac807d868427253c

**Module Membership**

Additionally, we will add columns describing the membership of each transcript to its module (as determined by our earlier run of WGCNA).

**DEG status**

A column will be added to denote DEGs for each pairwise comparison. 1 will signify DEGs, 0 will signify non-DEGs.

**Setup constants**
We'll keep a few things constant between runs

- The head of each table will be the same

- The Swiss-Prot database will be the same. It's also large, so we only want to read in once.

```{r}
#### Setup ----------------
# These will be used to build each full annotation table

# Set up header for outfmt 6
table_head <- c("query_seqid", "subj_seqid", "pident",
                "length", "mismatch", "gapopen",
                "qstart", "qend", "sstart",
                "send", "evalue", "bitscore")

# Read in full Swiss-ProtKB/Uniprot database
sprot_db <- read.delim(file = "../data/all_uniprot_info_inc_GOterms.tab",
                       header = TRUE)

# Rename:
#   Entry column to Accession_ID 
#   length to sprot.length 
sprot_db <- sprot_db %>%
  dplyr::rename(Accession_ID = Entry,
                sprot.length = Length)
```

**Write function**

This function will:
- Read in original BLAST table
- Join with Swiss-Prot database
- Read in module color and join
- Read in DEGs and join

Inputs:
- transcriptome: Should be kept in ../data/blast_tables/[transcriptome]_blast_table.txt
    - cbai_transcriptomev2.0: complete transcriptome
    - cbai_transcriptomev4.0: Tanner crab transcriptome
    - hemat_transcriptomev1.6: Parasite transcriptome
    
- out_name: Should match the more formal name as the transcriptome with no spaces 
    - if transcriptome is cbai_transcriptomev4.0, out_name = "Tanner_crab")
    
- sprot_db: Swiss-Prot database table. Should be already called above

- pairwise_compars: Relevant pairwise comparisons to include in table.
    - Use same notation as found in ../graphs/DESeq2_output/[transcriptome]/[compaison]_indiv/
        - ex: Elevated Day 0 vs. Day 2 is elev0_vs_elev2
    
```{r}
make_annot_table <- function(transcriptome, out_name, sprot_db, pairwise_compars){
  # Read in original BLASTx table
  annotation <- read.delim(file = paste0("../data/blast_tables/", 
                                         transcriptome, 
                                         "_blast_table.txt"),
                           header = FALSE,
                           col.names = table_head) 
  
  # Second column contains several relevant pieces of info
  # Split to extract relevant info (accession ID), which we'll use to merge
  # and then remove the "sp" column, which just reads "sp" for all entries
  annotation <- annotation %>%
    separate(col = subj_seqid,
             into = c("sp", "Accession_ID", "species"), 
             sep = "\\|") %>%
    select(-sp)
  
  # Left join annotation table with Swiss-Prot database by accession ID
  full_annot <- left_join(annotation, sprot_db, by = "Accession_ID")
  
  # Get a string of all files describing module membership
  mod_files <- list.files(path = paste0("../output/WGCNA_output/", 
                                        transcriptome, 
                                        "/all_crabs_no_filter/day_as_var/"),
                          pattern = "^GeneList",
                          full.names = TRUE)
  
  # Create empty matrix for adding module data
  modules <- matrix(ncol = 2, nrow = 0)
  
  # Loop through module names, adding each to table
  for (i in 1:length(mod_files)) {
    # Extract color of module for name
    mod_col <- unlist(str_split(mod_files[i], 
                                pattern = "GeneList-"))[2]
    mod_col <- unlist(str_split(mod_col,
                                pattern = ".txt"))[1]
    # Read in each module (one column of transcript IDs)
    mod <- read.delim(mod_files[i],
                      header = FALSE)
    
    # Add module color as column
    mod[, 2] <- rep(mod_col, times = nrow(mod))
    
    # Append to end of full module data matrix
    modules <- rbind(modules, mod)
  }
  
  # Now that it's full, rename columns of module matrix
  colnames(modules) <- c("query_seqid", "module_color")
  
  # Left join module membership data to full annotation table
  full_annot <- left_join(full_annot, modules, by = "query_seqid")
  
  # We have a number of files of DEGs. Each pairwise comparison will be a separate column.
  
  # Loop through pairwise comparisons
  for (i in 1:length(pairwise_compars)) {
    pairwise <- pairwise_compars[i]
    DEG_table <- read.delim(file = paste0("../graphs/DESeq2_output/",
                                          transcriptome,
                                          "/",
                                          pairwise,
                                          "_indiv/DEGlist_wcols.txt"))
    # Move rownames to columns, then remove all other columns
    DEG_table <- DEG_table %>%
      rownames_to_column(var = "query_seqid")
    DEG_table <- select(DEG_table, query_seqid)
    
    # Add a second column of ones, since all DEGs will be marked with a one
    DEG_table[ ,2] <- rep(1, times = nrow(DEG_table))
    
    # Rename column to show which pairwise comparison is being described
    colnames(DEG_table)[2] <- paste0("DEG_", pairwise)
    
    # Left join with full annotation table
    full_annot <- left_join(full_annot, DEG_table, by = "query_seqid")
    
  }
  
  # Write table to file
  write_delim(full_annot, 
            file = paste0("../paper/supp_files/annotation_tables/",
                          out_name, 
                          "_transcriptome.txt"),
            delim = "\t")
}
```

**Call function to create annotation tables**

Do this once for each transcriptome

```{r}
# Complete transcriptome
make_annot_table(transcriptome = "cbai_transcriptomev2.0",
                 out_name = "complete",
                 sprot_db = sprot_db,
                 pairwise_compars = c("elev0_vs_elev2",
                                      "low0_vs_low2",
                                      "amb0_vs_amb2",
                                      "amb0_vs_amb17"))

# Tanner crab transcriptome
make_annot_table(transcriptome = "cbai_transcriptomev4.0",
                 out_name = "Tanner_crab",
                 sprot_db = sprot_db,
                 pairwise_compars = c("elev0_vs_elev2",
                                      "low0_vs_low2",
                                      "amb0_vs_amb2",
                                      "amb0_vs_amb17"))

# Hematodinium transcriptome
make_annot_table(transcriptome = "hemat_transcriptomev1.6",
                 out_name = "parasite",
                 sprot_db = sprot_db,
                 pairwise_compars = c("elev0_vs_elev2",
                                      "low0_vs_low2",
                                      "amb0_vs_amb2",
                                      "amb0_vs_amb17"))
```



```{r}



#### Complete Transcriptome ---------------------------------

# Read in data
annotation <- read.delim(file = "../data/blast_tables/cbai_transcriptomev2.0_blast_table.txt",
                         header = FALSE,
                         col.names = table_head)

# Second column contains several relevant pieces of info
# Split to extract relevant info (accession ID), which we'll use to merge
# and then remove the "sp" column, which just reads "sp" for all entries
annotation <- annotation %>%
  separate(col = subj_seqid,
           into = c("sp", "Accession_ID", "species"), 
           sep = "\\|") %>%
  select(-sp)

# Left join annotation table with Swiss-Prot database by accession ID
full_annot <- left_join(annotation, sprot_db, by = "Accession_ID")

# Get a string of all files describing module membership
mod_files <- list.files(path = "../output/WGCNA_output/cbai_transcriptomev2.0/all_crabs_no_filter/day_as_var/",
                        pattern = "^GeneList",
                        full.names = TRUE)

# Create empty matrix for adding module data
modules <- matrix(ncol = 2, nrow = 0)

# Loop through module names, adding each to table
for (i in 1:length(mod_files)) {
  # Extract color of module for name
  mod_col <- unlist(str_split(mod_files[i], 
                       pattern = "GeneList-"))[2]
  mod_col <- unlist(str_split(mod_col,
                              pattern = ".txt"))[1]
  # Read in each module (one column of transcript IDs)
  mod <- read.delim(mod_files[i],
                    header = FALSE)
  
  # Add module color as column
  mod[, 2] <- rep(mod_col, times = nrow(mod))
  
  # Append to end of full module data matrix
  modules <- rbind(modules, mod)
}

# Now that it's full, rename columns of module matrix
colnames(modules) <- c("query_seqid", "module_color")

# Left join module membership data to full annotation table
full_annot <- left_join(full_annot, modules, by = "query_seqid")

# Extract table of DEGs

# Write table to file
write.csv(full_annot, file = "../paper/supp_files/annotation_tables/complete_transcriptome.csv",
          row.names = FALSE)

#### Tanner Crab Transcriptome ---------------------------------

# Read in data
annotation <- read.delim(file = "../data/blast_tables/cbai_transcriptomev4.0_blast_table.txt",
                         header = FALSE,
                         col.names = table_head)

# Second column contains several relevant pieces of info
# Split to extract relevant info (accession ID), which we'll use to merge
# and then remove the "sp" column, which just reads "sp" for all entries
annotation <- annotation %>%
  separate(col = subj_seqid,
           into = c("sp", "Accession_ID", "species"), 
           sep = "\\|") %>%
  select(-sp)

# Left join annotation table with Swiss-Prot database by accession ID
full_annot <- left_join(annotation, sprot_db, by = "Accession_ID")

# Get a string of all files describing module membership
mod_files <- list.files(path = "../output/WGCNA_output/cbai_transcriptomev4.0/all_crabs_no_filter/day_as_var/",
                        pattern = "^GeneList",
                        full.names = TRUE)

# Create empty matrix for adding module data
modules <- matrix(ncol = 2, nrow = 0)

# Loop through module names, adding each to table
for (i in 1:length(mod_files)) {
  # Extract color of module for name
  mod_col <- unlist(str_split(mod_files[i], 
                       pattern = "GeneList-"))[2]
  mod_col <- unlist(str_split(mod_col,
                              pattern = ".txt"))[1]
  # Read in each module (one column of transcript IDs)
  mod <- read.delim(mod_files[i],
                    header = FALSE)
  
  # Add module color as column
  mod[, 2] <- rep(mod_col, times = nrow(mod))
  
  # Append to end of full module data matrix
  modules <- rbind(modules, mod)
}

# Now that it's full, rename columns of module matrix
colnames(modules) <- c("query_seqid", "module_color")

# Left join module membership data to full annotation table
full_annot <- left_join(full_annot, modules, by = "query_seqid")

# Write table to file
write.csv(full_annot, file = "../paper/supp_files/annotation_tables/Tanner_crab_transcriptome.csv",
          row.names = FALSE)

#### Hematodinium Transcriptome ---------------------------------

# Read in data
annotation <- read.delim(file = "../data/blast_tables/hemat_transcriptomev1.6_blast_table.txt",
                         header = FALSE,
                         col.names = table_head)

# Second column contains several relevant pieces of info
# Split to extract relevant info (accession ID), which we'll use to merge
# and then remove the "sp" column, which just reads "sp" for all entries
annotation <- annotation %>%
  separate(col = subj_seqid,
           into = c("sp", "Accession_ID", "species"), 
           sep = "\\|") %>%
  select(-sp)

# Left join annotation table with Swiss-Prot database by accession ID
full_annot <- left_join(annotation, sprot_db, by = "Accession_ID")

# Get a string of all files describing module membership
mod_files <- list.files(path = "../output/WGCNA_output/hemat_transcriptomev1.6/all_crabs_no_filter/day_as_var/",
                        pattern = "^GeneList",
                        full.names = TRUE)

# Create empty matrix for adding module data
modules <- matrix(ncol = 2, nrow = 0)

# Loop through module names, adding each to table
for (i in 1:length(mod_files)) {
  # Extract color of module for name
  mod_col <- unlist(str_split(mod_files[i], 
                       pattern = "GeneList-"))[2]
  mod_col <- unlist(str_split(mod_col,
                              pattern = ".txt"))[1]
  # Read in each module (one column of transcript IDs)
  mod <- read.delim(mod_files[i],
                    header = FALSE)
  
  # Add module color as column
  mod[, 2] <- rep(mod_col, times = nrow(mod))
  
  # Append to end of full module data matrix
  modules <- rbind(modules, mod)
}

# Now that it's full, rename columns of module matrix
colnames(modules) <- c("query_seqid", "module_color")

# Left join module membership data to full annotation table
full_annot <- left_join(full_annot, modules, by = "query_seqid")

# Write table to file
write.csv(full_annot, file = "../paper/supp_files/annotation_tables/Hematodinium_transcriptome.csv",
          row.names = FALSE)
```




