---
title: "32_kallisto_to_deseq_to_accessionIDs"
author: "Aidan Coyle"
date: "Last compiled on `r format(Sys.time(), '%Y-%m-%d')`"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

Originally written 2021/03/10
Roberts Lab, UW-SAFS
Written for analysis of _Hematodinium_ differential gene expression

**IMPORTANT: Many code blocks have been set to eval = FALSE due to long runtimes. To re-run analysis, change those blocks to TRUE**

# Purpose:
This script utilizes a series of custom-built functions for the following: 
- running a Trinity matrix of kallisto transcript counts through DESeq2
- turning DESeq2 output into newline-separated file of UniProt accessions
- creating Venn diagrams of transcript IDs and accession IDs (not needed for future steps - can skip if not wanted)



```{r libraries, message = FALSE, warning=FALSE}
# Add all required libraries that are installed with install.packages() here
list.of.packages <- c("tidyverse", "plotly")
# Add all libraries that are installed using BiocManager here
bioconductor.packages <- c("apeglm", "DESeq2", "vsn")

# Install BiocManager if needed
if(!requireNamespace("BiocManager", quietly = TRUE)) install.packages("BiocManager")

# Get names of all required packages that aren't installed
new.packages <- list.of.packages[!(list.of.packages %in% installed.packages()[, "Package"])]
new.bioc.packages <- bioconductor.packages[!(bioconductor.packages %in% installed.packages()[, "Package"])]
# Install all new packages
if(length(new.packages)) install.packages(new.packages)
if(length(new.bioc.packages)) BiocManager::install(new.bioc.packages)

# Load all required libraries
all.packages <- c(list.of.packages, bioconductor.packages)
lapply(all.packages, FUN = function(X) {
  do.call("require", list(X))
})

# Functions are defined in hematodinium_analysis_functions.R
source("hematodinium_analysis_functions.R")
```

# Stage 1: Analyzing Trinity matrix from Kallisto output with DESeq2

Almost all analysis takes place in function deseq_analysis(). Documentation in function file. If attempting to redo a specific graph or table, the file DESeq_demonstration.R may be helpful

These blocks take quite a long time to run, so **in the markdown file, I have set eval to FALSE**. I recommend only re-running blocks that you are certain you want to run - again, run times are quite lengthy.

For all code blocks, **the order of rows in temperature and day values in exp_design should match the order of columns of the `kallisto.isoforms.counts.matrix` input file from Trinity for the pairwise comparison**. As an example, here's what the `exp_design` table should look like for hypothetical libraries A (day 0, lowered temp), B (day 2, ambient temp), and C (day 17, elevated temp) if the `kallisto.isoforms.counts.matrix` file columns are ordered A, B, C:

```{r example_table}
exp_design <- data.frame(Treatment = factor(c("Low", "Amb", "Elev")),
                         Day = factor(c(0, 2, 17)))

exp_design
```

As a further note, the pairwise comparisons performed here treat temperature as water temperature at time of sample, **not** treatment group. This means that Elevated Day 0 and Lowered Day 0 libraries are treated as **ambient-temperature libraries**, as samples were taken prior to any temperature changes. For all other samples, there are no such differences.

### Explanation of deseq_analysis():

A function to analyze Kallisto output with DESeq2. Will likely take quite long to run - there's some computationally-heavy steps involved

Takes 4 inputs

1. kallisto_path: path to matrix created by Trinity from Kallisto outputs

2. experiment_table: table with treatments in experiment. 

  + Order should match columns of matrix from Trinity

  + Example: Day and temperature data for libraries 118, 132, 178, 463, 481, 485 (in order)

  + exp_design <- data.frame(day = factor(c(0, 0, 0, 17, 17, 17)), temp = factor(c("Amb", "Amb", "Amb", "Amb", "Amb", "Amb")))

3. output_path: path to the output folder

4. variable: variable being examined. Used in creation of deseq object.

  + Should be 'day' or 'temp' (without quotes)


### Elevated Day 0 vs. Elevated Day 2, Individual Libraries Only
exp_design = day and temperature data for libraries 173, 072, 127, 272, 294, 280 (in order). 

```{r elev0vselev2, eval = FALSE}

# Rather than using the deseq_analysis() function, we will run that function directly.
# Due to low numbers of normalized counts, the function returns an error when ran.
# This will be avoided by removing the call to DESeq2::vst(), which performs the analysis 
# necessary for our PCA plot. Therefore, our output will lack a PCA plot, but will be 
# otherwise unchanged.


exp_design <- data.frame(Temperature = factor(c("Amb", "Amb", "Amb",
                                         "Elev", "Elev", "Elev")),
                         Day = factor(c(0, 0, 0, 2, 2, 2)))

kallisto_path <- "../output/kallisto_matrices/hemat_transcriptomev1.6/elev0_vs_elev2_indiv/kallisto.isoform.counts.matrix"
experiment_table <- exp_design
output_path <- "../graphs/DESeq2_output/hemat_transcriptomev1.6/elev0_vs_elev2_indiv"
variable <- "Temperature"

# Read in matrix created by Trinity from Kallisto outputs
data <- read.table(kallisto_path, header = TRUE,
                     sep = "\t")
# Rename first column
names(data)[1] <- "target_ID"
  
# Set row names equal to the first column
rownames(data) <- data$target_ID
  
# Remove the now-irrelevant first column
data <- data[,-1]
  
# Make sure everything looks okay
print("HEAD")
print(head(data))
  
print("STRUCTURE")
print(str(data))
  
# Round counts to integers - needed for DESeqDataSetFromMatrix()
data <- round(data, digits = 0)
  
# Rename rows to correspond to library numbers
rownames(experiment_table) <- colnames(data)
  
# Check that experiment_table appears to match columns with matrix from Trinity
print("EXPERIMENTAL DESIGN")
print(experiment_table)
  
# Create DESeq object that looks at effect of variable
deseq2.dds <- DESeqDataSetFromMatrix(countData = (data),
                                     colData = experiment_table,
                                     design = as.formula(paste0("~", variable)))
 
deseq2.dds <- DESeq(deseq2.dds)
  
#Look at results
deseq2.res <- results(deseq2.dds)
print("SUMMARY:")
print(summary(deseq2.res))
  
# Shrink LFC estimates - used in shrunken MA plot
lfcnames <- resultsNames(deseq2.dds)
print(lfcnames[2])
resLFC <- lfcShrink(deseq2.dds, coef = lfcnames[2], 
                    type = "apeglm")


# Look specifically at results w/ adjusted p-value < 0.05
deseq_res05 <- results(deseq2.dds, alpha = 0.05)
print("Number of DEGs (padj <= 0.05)")
print(sum(deseq_res05$padj < 0.05, na.rm = TRUE))
  
# Plot of full results, not shrunken
plotMA(deseq2.res, ylim = c(-28, 28))
dev.copy(png, file.path(output_path, "allres_MAplot.png"))
dev.off()
  
# Plot of full results, shrunken
plotMA(resLFC, ylim = c(-2, 2), main = "apeglm")
dev.copy(png, file.path(output_path, "allres_shrunken_MAplot.png"))
dev.off()
  
# Plot of res05 results, not shrunken
plotMA(deseq_res05, ylim = c(-20, 20))
dev.copy(png, file.path(output_path, "res05_MAplot.png"))
dev.off()
  
# Create a plot of Log2 fold change vs. normalized counts
deseq2_tmp <- deseq2.res
plot(deseq2_tmp$baseMean, deseq2_tmp$log2FoldChange, pch = 20,
     cex = 0.45, ylim = c(-28, 28), log = "x", col = "darkgray",
     main = paste("Differences by", variable, "(padj <= 0.005)"),
     xlab = "mean of normalized counts",
     ylab = "Log2 Fold Change")
# Get significant points, plot again so they're a diff color
deseq2_tmp.sig <- deseq2.res[!is.na(deseq2.res$padj) &
                               deseq2.res$padj <= 0.005, ]
points(deseq2_tmp.sig$baseMean, deseq2_tmp.sig$log2FoldChange,
       pch = 20, cex = 0.45, col = "red")
abline(h=c(-1,1), col = "blue")
dev.copy(png, file.path(output_path, "normalizedcts_v_log2foldchange.png"))
dev.off()
  
# Plot dispersion estimates
plotDispEsts(deseq2.dds)
dev.copy(png, file.path(output_path, "dispersion_estimates.png"))
dev.off()
  
# write all genes to table
write.table(deseq2_tmp, file.path(output_path, "AllGenes.txt"), 
            row.names = TRUE, col.names = FALSE, 
            quote = FALSE, sep = "\t")
write.table(deseq2_tmp, file.path(output_path, "AllGenes_wcols.txt"),
            row.names = TRUE, col.names = TRUE,
            quote = FALSE, sep = "\t")
  
# Write significant day-differing genes to table
write.table(deseq2_tmp.sig, file.path(output_path, "DEGlist.txt"),
            row.names = TRUE, col.names = FALSE, quote = FALSE,
            sep = "\t")
write.table(deseq2_tmp.sig, file.path(output_path, "DEGlist_wcols.txt"),
            row.names = TRUE, col.names = TRUE, quote = FALSE,
            sep = "\t")
  
```



### Ambient Day 2 vs. Elevated Day 2, Individual Libraries Only
exp_design = day and temperature data for libraries 359, 349, 334, 272, 294, 280 (in order)

```{r amb2vselev2, eval = FALSE}
exp_design <- data.frame(Temperature = factor(c("Ambient", "Ambient", "Ambient",
                                         "Elevated", "Elevated", "Elevated")),
                         Day = factor(c(2, 2, 2,
                                        2, 2, 2)))

deseq_analysis(kallisto_path = "../output/kallisto_matrices/hemat_transcriptomev1.6/amb2_vs_elev2_indiv/kallisto.isoform.counts.matrix",
               experiment_table = exp_design,
               output_path = "../graphs/DESeq2_output/hemat_transcriptomev1.6/amb2_vs_elev2_indiv",
               variable = "Temperature")
```

# Lowered Day 0 vs. Lowered Day 2, Individual Libraries Only

exp_design = day and temperature data for libraries 073, 151, 113, 221, 254, 222 (in order)

```{r low0vslow2, eval = FALSE}
# Due to low numbers of normalized counts, the function returns an error when ran.
# This will be avoided by removing the call to DESeq2::vst(), which performs the analysis 
# necessary for our PCA plot. Therefore, our output will lack a PCA plot, but will be 
# otherwise unchanged.

exp_design <- data.frame(Day = factor(c(0, 0, 0,
                                        2, 2, 2)))

kallisto_path <- "../output/kallisto_matrices/hemat_transcriptomev1.6/low0_vs_low2_indiv/kallisto.isoform.counts.matrix"
experiment_table <-  exp_design
output_path <- "../graphs/DESeq2_output/hemat_transcriptomev1.6/low0_vs_low2_indiv"
variable <- "Day"


# Read in matrix created by Trinity from Kallisto outputs
data <- read.table(kallisto_path, header = TRUE,
                     sep = "\t")
# Rename first column
names(data)[1] <- "target_ID"
  
# Set row names equal to the first column
rownames(data) <- data$target_ID
  
# Remove the now-irrelevant first column
data <- data[,-1]
  
# Make sure everything looks okay
print("HEAD")
print(head(data))
  
print("STRUCTURE")
print(str(data))
  
# Round counts to integers - needed for DESeqDataSetFromMatrix()
data <- round(data, digits = 0)
  
# Rename rows to correspond to library numbers
rownames(experiment_table) <- colnames(data)
  
# Check that experiment_table appears to match columns with matrix from Trinity
print("EXPERIMENTAL DESIGN")
print(experiment_table)
  
# Create DESeq object that looks at effect of variable
deseq2.dds <- DESeqDataSetFromMatrix(countData = (data),
                                     colData = experiment_table,
                                     design = as.formula(paste0("~", variable)))
 
deseq2.dds <- DESeq(deseq2.dds)
  
#Look at results
deseq2.res <- results(deseq2.dds)
print("SUMMARY:")
print(summary(deseq2.res))
  
# Shrink LFC estimates - used in shrunken MA plot
lfcnames <- resultsNames(deseq2.dds)
print(lfcnames[2])
resLFC <- lfcShrink(deseq2.dds, coef = lfcnames[2], 
                    type = "apeglm")


# Look specifically at results w/ adjusted p-value < 0.05
deseq_res05 <- results(deseq2.dds, alpha = 0.05)
print("Number of DEGs (padj <= 0.05)")
print(sum(deseq_res05$padj < 0.05, na.rm = TRUE))
  
# Plot of full results, not shrunken
plotMA(deseq2.res, ylim = c(-28, 28))
dev.copy(png, file.path(output_path, "allres_MAplot.png"))
dev.off()
  
# Plot of full results, shrunken
plotMA(resLFC, ylim = c(-2, 2), main = "apeglm")
dev.copy(png, file.path(output_path, "allres_shrunken_MAplot.png"))
dev.off()
  
# Plot of res05 results, not shrunken
plotMA(deseq_res05, ylim = c(-20, 20))
dev.copy(png, file.path(output_path, "res05_MAplot.png"))
dev.off()
  
# Create a plot of Log2 fold change vs. normalized counts
deseq2_tmp <- deseq2.res
plot(deseq2_tmp$baseMean, deseq2_tmp$log2FoldChange, pch = 20,
     cex = 0.45, ylim = c(-28, 28), log = "x", col = "darkgray",
     main = paste("Differences by", variable, "(padj <= 0.005)"),
     xlab = "mean of normalized counts",
     ylab = "Log2 Fold Change")
# Get significant points, plot again so they're a diff color
deseq2_tmp.sig <- deseq2.res[!is.na(deseq2.res$padj) &
                               deseq2.res$padj <= 0.005, ]
points(deseq2_tmp.sig$baseMean, deseq2_tmp.sig$log2FoldChange,
       pch = 20, cex = 0.45, col = "red")
abline(h=c(-1,1), col = "blue")
dev.copy(png, file.path(output_path, "normalizedcts_v_log2foldchange.png"))
dev.off()
  
# Plot dispersion estimates
plotDispEsts(deseq2.dds)
dev.copy(png, file.path(output_path, "dispersion_estimates.png"))
dev.off()
  
# write all genes to table
write.table(deseq2_tmp, file.path(output_path, "AllGenes.txt"), 
            row.names = TRUE, col.names = FALSE, 
            quote = FALSE, sep = "\t")
write.table(deseq2_tmp, file.path(output_path, "AllGenes_wcols.txt"),
            row.names = TRUE, col.names = TRUE,
            quote = FALSE, sep = "\t")
  
# Write significant day-differing genes to table
write.table(deseq2_tmp.sig, file.path(output_path, "DEGlist.txt"),
            row.names = TRUE, col.names = FALSE, quote = FALSE,
            sep = "\t")
write.table(deseq2_tmp.sig, file.path(output_path, "DEGlist_wcols.txt"),
            row.names = TRUE, col.names = TRUE, quote = FALSE,
            sep = "\t")
  
```

# Lowered Day 0 vs. Lowered Days 2 + 17, Individual Libraries Only

exp_design = day and temperature data for libraries 073, 151, 113, 221, 254, 222, 427, 445, 425 (in order)

```{r low0vslow217, eval = FALSE}
# Due to low numbers of normalized counts, the function returns an error when ran.
# This will be avoided by removing the call to DESeq2::vst(), which performs the analysis 
# necessary for our PCA plot. Therefore, our output will lack a PCA plot, but will be 
# otherwise unchanged.

exp_design <- data.frame(Temperature = factor(c("Ambient", "Ambient", "Ambient",
                                                "Decreased", "Decreased", "Decreased",
                                                "Decreased", "Decreased", "Decreased")),
                         Day = factor(c(0, 0, 0,
                                        2, 2, 2,
                                        17, 17, 17)))

kallisto_path <- "../output/kallisto_matrices/hemat_transcriptomev1.6/low0_vs_low217_indiv/kallisto.isoform.counts.matrix"
experiment_table <-  exp_design
output_path <- "../graphs/DESeq2_output/hemat_transcriptomev1.6/low0_vs_low217_indiv"
variable <- "Temperature"


# Read in matrix created by Trinity from Kallisto outputs
data <- read.table(kallisto_path, header = TRUE,
                     sep = "\t")
# Rename first column
names(data)[1] <- "target_ID"
  
# Set row names equal to the first column
rownames(data) <- data$target_ID
  
# Remove the now-irrelevant first column
data <- data[,-1]
  
# Make sure everything looks okay
print("HEAD")
print(head(data))
  
print("STRUCTURE")
print(str(data))
  
# Round counts to integers - needed for DESeqDataSetFromMatrix()
data <- round(data, digits = 0)
  
# Rename rows to correspond to library numbers
rownames(experiment_table) <- colnames(data)
  
# Check that experiment_table appears to match columns with matrix from Trinity
print("EXPERIMENTAL DESIGN")
print(experiment_table)
  
# Create DESeq object that looks at effect of variable
deseq2.dds <- DESeqDataSetFromMatrix(countData = (data),
                                     colData = experiment_table,
                                     design = as.formula(paste0("~", variable)))
 
deseq2.dds <- DESeq(deseq2.dds)
  
#Look at results
deseq2.res <- results(deseq2.dds)
print("SUMMARY:")
print(summary(deseq2.res))
  
# Shrink LFC estimates - used in shrunken MA plot
lfcnames <- resultsNames(deseq2.dds)
print(lfcnames[2])
resLFC <- lfcShrink(deseq2.dds, coef = lfcnames[2], 
                    type = "apeglm")


# Look specifically at results w/ adjusted p-value < 0.05
deseq_res05 <- results(deseq2.dds, alpha = 0.05)
print("Number of DEGs (padj <= 0.05)")
print(sum(deseq_res05$padj < 0.05, na.rm = TRUE))
  
# Plot of full results, not shrunken
plotMA(deseq2.res, ylim = c(-28, 28))
dev.copy(png, file.path(output_path, "allres_MAplot.png"))
dev.off()
  
# Plot of full results, shrunken
plotMA(resLFC, ylim = c(-2, 2), main = "apeglm")
dev.copy(png, file.path(output_path, "allres_shrunken_MAplot.png"))
dev.off()
  
# Plot of res05 results, not shrunken
plotMA(deseq_res05, ylim = c(-20, 20))
dev.copy(png, file.path(output_path, "res05_MAplot.png"))
dev.off()
  
# Create a plot of Log2 fold change vs. normalized counts
deseq2_tmp <- deseq2.res
plot(deseq2_tmp$baseMean, deseq2_tmp$log2FoldChange, pch = 20,
     cex = 0.45, ylim = c(-28, 28), log = "x", col = "darkgray",
     main = paste("Differences by", variable, "(padj <= 0.005)"),
     xlab = "mean of normalized counts",
     ylab = "Log2 Fold Change")
# Get significant points, plot again so they're a diff color
deseq2_tmp.sig <- deseq2.res[!is.na(deseq2.res$padj) &
                               deseq2.res$padj <= 0.005, ]
points(deseq2_tmp.sig$baseMean, deseq2_tmp.sig$log2FoldChange,
       pch = 20, cex = 0.45, col = "red")
abline(h=c(-1,1), col = "blue")
dev.copy(png, file.path(output_path, "normalizedcts_v_log2foldchange.png"))
dev.off()
  
# Plot dispersion estimates
plotDispEsts(deseq2.dds)
dev.copy(png, file.path(output_path, "dispersion_estimates.png"))
dev.off()
  
# write all genes to table
write.table(deseq2_tmp, file.path(output_path, "AllGenes.txt"), 
            row.names = TRUE, col.names = FALSE, 
            quote = FALSE, sep = "\t")
write.table(deseq2_tmp, file.path(output_path, "AllGenes_wcols.txt"),
            row.names = TRUE, col.names = TRUE,
            quote = FALSE, sep = "\t")
  
# Write significant day-differing genes to table
write.table(deseq2_tmp.sig, file.path(output_path, "DEGlist.txt"),
            row.names = TRUE, col.names = FALSE, quote = FALSE,
            sep = "\t")
write.table(deseq2_tmp.sig, file.path(output_path, "DEGlist_wcols.txt"),
            row.names = TRUE, col.names = TRUE, quote = FALSE,
            sep = "\t")
  
```

# Lowered Day 0 + Ambient Day 0/2/17 vs. Lowered Day 2/17, Individual

exp_design = day and temperature data for libraries 073, 151, 113, 178, 118, 132, 359, 349, 334, 463, 481, 485, 221, 254, 222, 427, 445, 425

```{r low0amb0217vslow217, eval = FALSE}
exp_design <- data.frame(Temperature = factor(c("Ambient", "Ambient", "Ambient",
                                                "Ambient", "Ambient", "Ambient",
                                                "Ambient", "Ambient", "Ambient",
                                                "Ambient", "Ambient", "Ambient",
                                                "Decreased", "Decreased", "Decreased",
                                                "Decreased", "Decreased", "Decreased")),
                         Day = factor(c(0, 0, 0,
                                        0, 0, 0,
                                        2, 2, 2,
                                        17, 17, 17,
                                        2, 2, 2,
                                        17, 17, 17)))

deseq_analysis(kallisto_path = "../output/kallisto_matrices/hemat_transcriptomev1.6/low0_amb0217_vs_low217_indiv/kallisto.isoform.counts.matrix",
               experiment_table = exp_design,
               output_path = "../graphs/DESeq2_output/hemat_transcriptomev1.6/low0_amb0217_vs_low217_indiv",
               variable = "Temperature")
```

# Lowered Day 2 + 17 vs. Ambient Day 2 + 17, Individual

exp_design = day and temperature data for libraries 221, 254, 222, 427, 445, 425, 359, 349, 334, 463, 481, 485 (in order)

```{r low217vsamb217, eval = FALSE}
exp_design <- data.frame(Temperature = factor(c("Decreased", "Decreased", "Decreased",
                                                "Decreased", "Decreased", "Decreased",
                                                "Ambient", "Ambient", "Ambient",
                                                "Ambient", "Ambient", "Ambient")),
                         Day = factor(c(2, 2, 2,
                                        17, 17, 17,
                                        2, 2, 2,
                                        17, 17, 17)))

deseq_analysis(kallisto_path = "../output/kallisto_matrices/hemat_transcriptomev1.6/low217_vs_amb217_indiv/kallisto.isoform.counts.matrix",
               experiment_table = exp_design,
               output_path = "../graphs/DESeq2_output/hemat_transcriptomev1.6/low217_vs_amb217_indiv",
               variable = "Temperature")
```

### Decreased Day 0 vs. Ambient Day 0, Individual Libraries Only
exp_design = day and temperature data for libraries 073, 151, 113, 178, 118, 132 (in order)

Note: In this case, "Temperature" stands for TREATMENT temperature, not ACTUAL temperature (in all others, it's the actual temperature). However, this is a test for differences in base infection rate for treatment groups.

```{r low0vsamb0, eval = FALSE}
exp_design <- data.frame(Temperature = factor(c("Decreased", "Decreased", "Decreased",
                                         "Ambient", "Ambient", "Ambient")))

deseq_analysis(kallisto_path = "../output/kallisto_matrices/hemat_transcriptomev1.6/low0_vs_amb0_indiv/kallisto.isoform.counts.matrix",
               experiment_table = exp_design,
               output_path = "../graphs/DESeq2_output/hemat_transcriptomev1.6/low0_vs_amb0_indiv",
               variable = "Temperature")
```



### Ambient Day 0 vs. Ambient Day 2, Individual Libraries Only
exp_design = day and temperature data for libraries 178, 118, 132, 359, 349, 334 (in order)

```{r amb0vsamb2, eval = FALSE}
exp_design <- data.frame(Day = factor(c("Day 0", "Day 0", "Day 0", 
                                        "Day 2", "Day 2", "Day 2")),
                         Temperature = factor(c("Ambient", "Ambient", "Ambient",
                                         "Ambient", "Ambient", "Ambient")))

deseq_analysis(kallisto_path = "../output/kallisto_matrices/hemat_transcriptomev1.6/amb0_vs_amb2_indiv/kallisto.isoform.counts.matrix",
               experiment_table = exp_design,
               output_path = "../graphs/DESeq2_output/hemat_transcriptomev1.6/amb0_vs_amb2_indiv",
               variable = "Day")
```

### Ambient Day 0 vs. Ambient Day 17, Individual Libraries Only
exp_design = day and temperature data for libraries 178, 118, 132, 463, 481, 485 (in order)

```{r amb0vsamb17, eval = FALSE}
exp_design <- data.frame(Day = factor(c("Day 0", "Day 0", "Day 0", 
                                        "Day 17", "Day 17", "Day 17")))

deseq_analysis(kallisto_path = "../output/kallisto_matrices/hemat_transcriptomev1.6/amb0_vs_amb17_indiv/kallisto.isoform.counts.matrix",
               experiment_table = exp_design,
               output_path = "../graphs/DESeq2_output/hemat_transcriptomev1.6/amb0_vs_amb17_indiv",
               variable = "Day")
```


### All Individual Libraries
exp_design = day and temperature data for libraries 178, 359, 463, 118, 349, 481, 132, 334, 485, 073, 221, 427, 151, 254, 445, 113, 222, 425, 173, 272, 072, 294, 127, 280. 

That corresponds to Crab A Day 0, Crab A Day 2, Crab A Day 17, Crab B Day 0, Crab B Day 2... for crabs A-I in alphabetical order

#### Temperature as variable

Using temperature at time of sampling, not treatment  group

```{r all_libs_temp, eval = FALSE}
exp_design <- data.frame(Temperature = factor(c("Ambient", "Ambient", "Ambient",
                                         "Ambient", "Ambient", "Ambient",
                                         "Ambient", "Ambient", "Ambient",
                                         "Ambient", "Decreased", "Decreased",
                                         "Ambient", "Decreased", "Decreased",
                                         "Ambient", "Decreased", "Decreased",
                                         "Ambient", "Elevated", "Ambient",
                                         "Elevated", "Ambient", "Elevated")),
                         Day = factor(c("Day 0", "Day 2", "Day 17",
                                        "Day 0", "Day 2", "Day 17",
                                        "Day 0", "Day 2", "Day 17",
                                        "Day 0", "Day 2", "Day 17",
                                        "Day 0", "Day 2", "Day 17", 
                                        "Day 0", "Day 2", "Day 17", 
                                        "Day 0", "Day 2", "Day 0", 
                                        "Day 2", "Day 0", "Day 2")),
                         ID = factor(c(rep("Crab A", times = 3),
                                         rep("Crab B", times = 3),
                                         rep("Crab C", times = 3),
                                         rep("Crab D", times = 3),
                                         rep("Crab E", times = 3),
                                         rep("Crab F", times = 3),
                                         rep("Crab G", times = 2),
                                         rep("Crab H", times = 2),
                                         rep("Crab I", times = 2))))

deseq_analysis(kallisto_path = "../output/kallisto_matrices/hemat_transcriptomev1.6/all_indiv_libraries/kallisto.isoform.counts.matrix",
               experiment_table = exp_design,
               output_path = "../graphs/DESeq2_output/hemat_transcriptomev1.6/all_indiv_libraries/temp_as_var/",
               variable = "Temperature")
```


#### Crab as variable


```{r all_libs_crab, eval = FALSE}
exp_design <- data.frame(Temperature = factor(c("Ambient", "Ambient", "Ambient",
                                         "Ambient", "Ambient", "Ambient",
                                         "Ambient", "Ambient", "Ambient",
                                         "Ambient", "Decreased", "Decreased",
                                         "Ambient", "Decreased", "Decreased",
                                         "Ambient", "Decreased", "Decreased",
                                         "Ambient", "Elevated", "Ambient",
                                         "Elevated", "Ambient", "Elevated")),
                         Day = factor(c("Day 0", "Day 2", "Day 17",
                                        "Day 0", "Day 2", "Day 17",
                                        "Day 0", "Day 2", "Day 17",
                                        "Day 0", "Day 2", "Day 17",
                                        "Day 0", "Day 2", "Day 17", 
                                        "Day 0", "Day 2", "Day 17", 
                                        "Day 0", "Day 2", "Day 0", 
                                        "Day 2", "Day 0", "Day 2")),
                         Individual = factor(c(rep("Crab A", times = 3),
                                         rep("Crab B", times = 3),
                                         rep("Crab C", times = 3),
                                         rep("Crab D", times = 3),
                                         rep("Crab E", times = 3),
                                         rep("Crab F", times = 3),
                                         rep("Crab G", times = 2),
                                         rep("Crab H", times = 2),
                                         rep("Crab I", times = 2))))

deseq_analysis(kallisto_path = "../output/kallisto_matrices/hemat_transcriptomev1.6/all_indiv_libraries/kallisto.isoform.counts.matrix",
               experiment_table = exp_design,
               output_path = "../graphs/DESeq2_output/hemat_transcriptomev1.6/all_indiv_libraries/crab_as_var/",
               variable = "Individual")
```

#### Day as variable


```{r all_libs_day, eval = FALSE}
exp_design <- data.frame(Temperature = factor(c("Ambient", "Ambient", "Ambient",
                                         "Ambient", "Ambient", "Ambient",
                                         "Ambient", "Ambient", "Ambient",
                                         "Ambient", "Decreased", "Decreased",
                                         "Ambient", "Decreased", "Decreased",
                                         "Ambient", "Decreased", "Decreased",
                                         "Ambient", "Elevated", "Ambient",
                                         "Elevated", "Ambient", "Elevated")),
                         Day = factor(c("Day 0", "Day 2", "Day 17",
                                        "Day 0", "Day 2", "Day 17",
                                        "Day 0", "Day 2", "Day 17",
                                        "Day 0", "Day 2", "Day 17",
                                        "Day 0", "Day 2", "Day 17", 
                                        "Day 0", "Day 2", "Day 17", 
                                        "Day 0", "Day 2", "Day 0", 
                                        "Day 2", "Day 0", "Day 2")),
                         Individual = factor(c(rep("Crab A", times = 3),
                                         rep("Crab B", times = 3),
                                         rep("Crab C", times = 3),
                                         rep("Crab D", times = 3),
                                         rep("Crab E", times = 3),
                                         rep("Crab F", times = 3),
                                         rep("Crab G", times = 2),
                                         rep("Crab H", times = 2),
                                         rep("Crab I", times = 2))))

deseq_analysis(kallisto_path = "../output/kallisto_matrices/hemat_transcriptomev1.6/all_indiv_libraries/kallisto.isoform.counts.matrix",
               experiment_table = exp_design,
               output_path = "../graphs/DESeq2_output/hemat_transcriptomev1.6/all_indiv_libraries/day_as_var/",
               variable = "Day")
```

#### Infection as variable


```{r all_libs_inf, eval = FALSE}
exp_design <- data.frame(Temperature = factor(c("Ambient", "Ambient", "Ambient",
                                                "Ambient", "Ambient", "Ambient",
                                                "Ambient", "Ambient", "Ambient",
                                                "Ambient", "Decreased", "Decreased",
                                                "Ambient", "Decreased", "Decreased",
                                                "Ambient", "Decreased", "Decreased",
                                                "Ambient", "Elevated", "Ambient",
                                                "Elevated", "Ambient", "Elevated")),
                         Individual = factor(c(rep("Crab A", times = 3),
                                       rep("Crab B", times = 3),
                                       rep("Crab C", times = 3),
                                       rep("Crab D", times = 3),
                                       rep("Crab E", times = 3),
                                       rep("Crab F", times = 3),
                                       rep("Crab G", times = 2),
                                       rep("Crab H", times = 2),
                                       rep("Crab I", times = 2))),
                         cPCR = factor(c(rep("Infected", times = 9),
                                         rep("Uninfected", times = 3),
                                         rep("Infected", times = 3),
                                         rep("Uninfected", times = 3),
                                         rep("Infected", times = 6))))

deseq_analysis(kallisto_path = "../output/kallisto_matrices/hemat_transcriptomev1.6/all_indiv_libraries/kallisto.isoform.counts.matrix",
               experiment_table = exp_design,
               output_path = "../graphs/DESeq2_output/hemat_transcriptomev1.6/all_indiv_libraries/infection_as_var/",
               variable = "cPCR")
```

#### Hematodinium infection level as variable

Hematodinium infection level was quantified using qPCR at a single time point for each crab. Levels can be found at [this link](https://github.com/crmateo/crab-capstone/blob/main/data/hematqpcr_crabRNA.csv).

The middle number in Sample ID corresponds to the Sample IDs used in this script (i.e. 6203_113_9  = Sample 113). As another note, the last digit corresponds to the experiment day. However, rather than counting from the first day of temperature manipulation, this counts from the first day the crabs were present in the tank. Essentially, it includes the 9-day acclimation period. Therefore, _9 = Day 0, _12 = Day 2, and _26 = Day 17

qPCR quantification (units presumed to be ng/ul)

Crab A = 1119

Crab B = 105477

Crab C = 333000

Crab D = 380502

Crab E = 1550

Crab F = 3069

Crab G = 4515

Crab H = 119500

Crab I = 211000





```{r all_libs_inf_status, eval = FALSE}
exp_design <- data.frame(Temperature = factor(c("Ambient", "Ambient", "Ambient",
                                         "Ambient", "Ambient", "Ambient",
                                         "Ambient", "Ambient", "Ambient",
                                         "Ambient", "Decreased", "Decreased",
                                         "Ambient", "Decreased", "Decreased",
                                         "Ambient", "Decreased", "Decreased",
                                         "Ambient", "Elevated", "Ambient",
                                         "Elevated", "Ambient", "Elevated")),
                         Day = factor(c("Day 0", "Day 2", "Day 17",
                                        "Day 0", "Day 2", "Day 17",
                                        "Day 0", "Day 2", "Day 17",
                                        "Day 0", "Day 2", "Day 17",
                                        "Day 0", "Day 2", "Day 17", 
                                        "Day 0", "Day 2", "Day 17", 
                                        "Day 0", "Day 2", "Day 0", 
                                        "Day 2", "Day 0", "Day 2")),
                         Infection = factor(c(rep("Light", times = 3),
                                               rep("Heavy", times = 9),
                                               rep("Light", times = 8),
                                               rep("Heavy", times = 4))))

deseq_analysis(kallisto_path = "../output/kallisto_matrices/hemat_transcriptomev1.6/all_indiv_libraries/kallisto.isoform.counts.matrix",
               experiment_table = exp_design,
               output_path = "../graphs/DESeq2_output/hemat_transcriptomev1.6/all_indiv_libraries/hemat_level_as_var/",
               variable = "Infection")
```

### All Samples

The purpose of this section is to create a PCA of all libraries. Each point would be one library. This lets us visualize what variables are influencing variance.

Because of this, we'll actually run DESeq more directly rather than calling the deseq_analysis() function - no need for a lot of the typical DESeq2 outputs. That'll also let us use ggplotly() to create a more interactive (and thus easier to visualize) PCA.

After that, we'll also calculate the plasticity by getting the data from our PCA and calculating the mean distance between each point (using PC1/PC2) and the centroid of that crab.

```{r}
# Create experimental table. Again, order should correspond to order of columns in our data file

exp_design <- data.frame(Temperature = factor(c("Ambient", "Ambient", "Ambient",
                                         "Ambient", "Ambient", "Ambient",
                                         "Ambient", "Ambient", "Ambient",
                                         "Ambient", "Decreased", "Decreased",
                                         "Ambient", "Decreased", "Decreased",
                                         "Ambient", "Decreased", "Decreased",
                                         "Ambient", "Elevated", "Ambient",
                                         "Elevated", "Ambient", "Elevated")),
                         Day = factor(c("Day 0", "Day 2", "Day 17",
                                        "Day 0", "Day 2", "Day 17",
                                        "Day 0", "Day 2", "Day 17",
                                        "Day 0", "Day 2", "Day 17",
                                        "Day 0", "Day 2", "Day 17", 
                                        "Day 0", "Day 2", "Day 17", 
                                        "Day 0", "Day 2", "Day 0", 
                                        "Day 2", "Day 0", "Day 2")),
                         Infection = factor(c(rep("Light", times = 3),
                                                rep("Heavy", times = 9),
                                                rep("Light", times = 8),
                                                rep("Heavy", times = 4))),
                         Individual = factor(c(rep("Crab A", times = 3),
                                       rep("Crab B", times = 3),
                                       rep("Crab C", times = 3),
                                       rep("Crab D", times = 3),
                                       rep("Crab E", times = 3),
                                       rep("Crab F", times = 3),
                                       rep("Crab G", times = 2),
                                       rep("Crab H", times = 2),
                                       rep("Crab I", times = 2))),
                         cPCR = factor(c(rep("Infected", times = 9),
                                              rep("Uninfected", times = 3),
                                              rep("Infected", times = 3),
                                              rep("Uninfected", times = 3),
                                              rep("Infected", times = 6))))

# Specify variable
variable <- "Individual"

# Read in data
data <- read.table("../output/kallisto_matrices/hemat_transcriptomev1.6/all_indiv_libraries/kallisto.isoform.counts.matrix", header = TRUE, sep = "\t")

# Rename first column
names(data)[1] <- "target_ID"

# Set row names equal to the first column
rownames(data) <- data$target_ID

# Remove the now-irrelevant first column
data <- data[,-1]

# Make sure everything looks okay
print("HEAD")
head(data)

print("STRUCTURE")
print(str(data))

# Round counts to integers - needed for DESeqDataSetFromMatrix()
data <- round(data, digits = 0)

# Rename rows to correspond to library numbers
rownames(exp_design) <- colnames(data)

# Check that experiment_table appears to match columns with matrix from Trinity
print("EXPERIMENTAL DESIGN")
print(exp_design)
  
# Create DESeq object that looks at effect of variable
deseq2.dds <- DESeqDataSetFromMatrix(countData = (data),
                                     colData = exp_design,
                                     design = as.formula("~Individual"))

deseq2.dds <- DESeq(deseq2.dds)

#Look at results
deseq2.res <- results(deseq2.dds)
print("SUMMARY:")
print(summary(deseq2.res))

# Transform values for PCA plot
vsd <- vst(deseq2.dds, blind = FALSE)

colData(vsd)

# Plot PCA

# Use plotPCA function to create the plot
pcaData <- DESeq2::plotPCA(vsd, intgroup=variable, returnData=TRUE)
# Round the percent variance for the labels
percentVar <- round(100 * attr(pcaData, "percentVar"))
# Create the actual plot 
ggplot(pcaData, aes(PC1, PC2, color = pcaData[, variable])) +
  geom_point(size=3) + 
  ggtitle("PCA of all crabs for hemat_transcriptomev1.6") +
  scale_color_discrete(variable) +
  xlab(paste0("PC1: ", percentVar[1], "% variance")) +
  ylab(paste0("PC2: ", percentVar[2], "% variance")) +
  coord_fixed()

rv <- rowVars(assay(vsd))
select <- order(rv, decreasing = TRUE)[seq_len(min(500, 
                                                   length(rv)))]
pca <- prcomp(t(assay(vsd)[select, ]))

# Find the points of each PCA
points <- pca[["x"]]

# Change rownames from sample ID to crab ID
rownames(points) <- exp_design$Individual

# Eliminate all columns except PC1 and PC2
points <- points[,1:2]

# Get list of all rownames used (should be all unique crab IDs)
crabs <- unique(rownames(points))

# Initialize a blank dataframe
means <- data.frame(crabs, rep(NA, times = length(crabs)), rep(NA, times = length(crabs)), rep(NA, times = length(crabs)))

# Rename the columns of the blank dataframe
colnames(means) <- c("crabID", "PC1_mean", "PC2_mean", "mean_dist_from_centroid")

for (i in 1:length(crabs)) {
  # Get values that match that particular crab
  crab_pts <- points[rownames(points) == crabs[i], ]
  
  # Get mean of each column. This is the coordinates of the centroid
  means[i, 2] <- mean(crab_pts[, 1])
  means[i, 3] <- mean(crab_pts[, 2])
  
  # Initialize blank vector that's the same length as the number of values that match the crab in question
  dist_vec <- rep(NA, length = nrow(crab_pts))
  
  for (j in 1:nrow(crab_pts)) {
    # Calculate the Euclidean distance from each point to the centroid, use to fill the blank vector
    dist_vec[j] <- sqrt((crab_pts[j,1] - means[i, 2])^2 + (crab_pts[j,2] - means[i, 3])^2)
    
  }
  # Take the mean of the Euclidean distances and put that in Column 4 of the means table
  means[i,4] <- mean(dist_vec)
}

# Remove the middle 2 columns, since we really just want mean distance in the final table
means <- select(means, -c(PC1_mean, PC2_mean))


write.table(means, file = "../graphs/DESeq2_output/hemat_transcriptomev1.6/all_indiv_libraries/PCA_plasticity.txt", row.names = FALSE)
```




# Stage 2: Turning Transcript IDs Into Accession IDs

The previous step produced four tables for each pairwise comparison, as follows. Each table has the same columns: transcript ID, base mean, log2 fold change, lfcSE, stat, p-value, p-adjusted 

1. Data for all differentially-expressed transcripts (DEGlist.txt)

2. Same as 1 but with headers (DEGlist_wcols.txt)

3. Data for all genes (AllGenes.txt)

4. Same as 3, but with headers (AllGenes_wcols.txt)

This  step turns those files (specifically 2 and 4) and turns them into a newline-separated file of UniProt accession IDs by cross-referencing with the BLASTx table for the transcriptome used as an index for kallisto.

### Explanation of transcripts_to_geneIDs:

Turns DESeq2 output to UniProt accessions

Takes three inputs:

1. deseq_filepath: leads to file containing gene list from DESeq2

2. blast_filepath: path that leads to transcript ID/gene ID table

3. output_path: path to new newline-separated file of UniProt accession IDs

### UniProt accession IDs for differentially-expressed transcripts

```{r allDEGs}

# Elevated Day 0 vs. Elevated Day 2
transcripts_to_geneIDs(deseq_filepath =  "../graphs/DESeq2_output/hemat_transcriptomev1.6/elev0_vs_elev2_indiv/DEGlist_wcols.txt", 
                       blast_filepath =  "../output/BLASTs/uniprot_swissprot/hemat1.6_blastxres.tab",
                       output_path =   "../output/accession_n_GOids/hemat_transcriptomev1.6/DEG_IDs/elev0_vs_elev2_indiv_DEG_IDs.txt")

# Ambient Day 2 vs. Elevated Day 2
transcripts_to_geneIDs(deseq_filepath = "../graphs/DESeq2_output/hemat_transcriptomev1.6/amb2_vs_elev2_indiv/DEGlist_wcols.txt", 
                       blast_filepath =  "../output/BLASTs/uniprot_swissprot/hemat1.6_blastxres.tab",
                       output_path = "../output/accession_n_GOids/hemat_transcriptomev1.6/DEG_IDs/amb2_vs_elev2_indiv_DEG_IDs.txt")

# Ambient Day 0 vs. Ambient Day 2
transcripts_to_geneIDs(deseq_filepath = "../graphs/DESeq2_output/hemat_transcriptomev1.6/amb0_vs_amb2_indiv/DEGlist_wcols.txt", 
                       blast_filepath =  "../output/BLASTs/uniprot_swissprot/hemat1.6_blastxres.tab",
                       output_path = "../output/accession_n_GOids/hemat_transcriptomev1.6/DEG_IDs/amb0_vs_amb2_indiv_DEG_IDs.txt")

# Ambient Day 0 vs. Ambient Day 17
transcripts_to_geneIDs(deseq_filepath = "../graphs/DESeq2_output/hemat_transcriptomev1.6/amb0_vs_amb17_indiv/DEGlist_wcols.txt", 
                       blast_filepath = "../output/BLASTs/uniprot_swissprot/hemat1.6_blastxres.tab",
                       output_path = "../output/accession_n_GOids/hemat_transcriptomev1.6/DEG_IDs/amb0_vs_amb17_indiv_DEG_IDs.txt")

# Lowered Day 0 vs. Lowered Day 2
transcripts_to_geneIDs(deseq_filepath = "../graphs/DESeq2_output/hemat_transcriptomev1.6/low0_vs_low2_indiv/DEGlist_wcols.txt", 
                       blast_filepath = "../output/BLASTs/uniprot_swissprot/hemat1.6_blastxres.tab",
                       output_path = "../output/accession_n_GOids/hemat_transcriptomev1.6/DEG_IDs/low0_vs_low2_indiv_DEG_IDs.txt")

# Lowered Day 0 vs. Lowered Day 2 + 17
transcripts_to_geneIDs(deseq_filepath = "../graphs/DESeq2_output/hemat_transcriptomev1.6/low0_vs_low217_indiv/DEGlist_wcols.txt", 
                       blast_filepath = "../output/BLASTs/uniprot_swissprot/hemat1.6_blastxres.tab",
                       output_path = "../output/accession_n_GOids/hemat_transcriptomev1.6/DEG_IDs/low0_vs_low217_indiv_DEG_IDs.txt")

# Lowered Day 0 + Ambient Day 0 + 2 + 17 vs. Lowered Day 2 + 17
transcripts_to_geneIDs(deseq_filepath = "../graphs/DESeq2_output/hemat_transcriptomev1.6/low0_amb0217_vs_low217_indiv/DEGlist_wcols.txt", 
                       blast_filepath = "../output/BLASTs/uniprot_swissprot/hemat1.6_blastxres.tab",
                       output_path = "../output/accession_n_GOids/hemat_transcriptomev1.6/DEG_IDs/low0_amb0217_vs_low217_indiv_DEG_IDs.txt")

# Lowered Day 2 + 17 vs. Ambient Day 2+17
transcripts_to_geneIDs(deseq_filepath = "../graphs/DESeq2_output/hemat_transcriptomev1.6/low217_vs_amb217_indiv/DEGlist_wcols.txt", 
                       blast_filepath = "../output/BLASTs/uniprot_swissprot/hemat1.6_blastxres.tab",
                       output_path = "../output/accession_n_GOids/hemat_transcriptomev1.6/DEG_IDs/low217_vs_amb217_indiv_DEG_IDs.txt")

# Lowered Day 0 vs. Ambient Day 0
transcripts_to_geneIDs(deseq_filepath = "../graphs/DESeq2_output/hemat_transcriptomev1.6/low0_vs_amb0_indiv/DEGlist_wcols.txt", 
                       blast_filepath = "../output/BLASTs/uniprot_swissprot/hemat1.6_blastxres.tab",
                       output_path = "../output/accession_n_GOids/hemat_transcriptomev1.6/DEG_IDs/low0_vs_amb0_indiv_DEG_IDs.txt")
```

### UniProt accession IDs for all transcripts

```{r allgenes}
# Elevated Day 0 vs. Elevated Day 2
transcripts_to_geneIDs(deseq_filepath = "../graphs/DESeq2_output/hemat_transcriptomev1.6/elev0_vs_elev2_indiv/AllGenes_wcols.txt", 
                       blast_filepath =  "../output/BLASTs/uniprot_swissprot/hemat1.6_blastxres.tab",
                       output_path = "../output/accession_n_GOids/hemat_transcriptomev1.6/allgenes_IDs/elev0_vs_elev2_indiv_All_GeneIDs.txt")

# Ambient Day 2 vs. Elevated Day 2
transcripts_to_geneIDs(deseq_filepath = "../graphs/DESeq2_output/hemat_transcriptomev1.6/amb2_vs_elev2_indiv/AllGenes_wcols.txt", 
                       blast_filepath =  "../output/BLASTs/uniprot_swissprot/hemat1.6_blastxres.tab",
                       output_path = "../output/accession_n_GOids/hemat_transcriptomev1.6/allgenes_IDs/amb2_vs_elev2_indiv_All_GeneIDs.txt")

# Ambient Day 0 vs. Ambient Day 2
transcripts_to_geneIDs(deseq_filepath = "../graphs/DESeq2_output/hemat_transcriptomev1.6/amb0_vs_amb2_indiv/AllGenes_wcols.txt", 
                       blast_filepath =  "../output/BLASTs/uniprot_swissprot/hemat1.6_blastxres.tab",
                       output_path = "../output/accession_n_GOids/hemat_transcriptomev1.6/allgenes_IDs/amb0_vs_amb2_indiv_All_GeneIDs.txt")

# Ambient Day 0 vs. Ambient Day 17
transcripts_to_geneIDs(deseq_filepath = "../graphs/DESeq2_output/hemat_transcriptomev1.6/amb0_vs_amb17_indiv/AllGenes_wcols.txt", 
                       blast_filepath = "../output/BLASTs/uniprot_swissprot/hemat1.6_blastxres.tab",
                       output_path = "../output/accession_n_GOids/hemat_transcriptomev1.6/allgenes_IDs/amb0_vs_amb17_indiv_All_GeneIDs.txt")

# Lowered Day 0 vs. Lowered Day 2
transcripts_to_geneIDs(deseq_filepath = "../graphs/DESeq2_output/hemat_transcriptomev1.6/low0_vs_low2_indiv/AllGenes_wcols.txt", 
                       blast_filepath = "../output/BLASTs/uniprot_swissprot/hemat1.6_blastxres.tab",
                       output_path = "../output/accession_n_GOids/hemat_transcriptomev1.6/allgenes_IDs/low0_vs_low2_indiv_All_GeneIDs.txt")

# Lowered Day 0 vs. Lowered Day 2 + 17
transcripts_to_geneIDs(deseq_filepath = "../graphs/DESeq2_output/hemat_transcriptomev1.6/low0_vs_low217_indiv/AllGenes_wcols.txt", 
                       blast_filepath = "../output/BLASTs/uniprot_swissprot/hemat1.6_blastxres.tab",
                       output_path = "../output/accession_n_GOids/hemat_transcriptomev1.6/allgenes_IDs/low0_vs_low217_indiv_All_GeneIDs.txt")

# Lowered Day 0 + Ambient Day 0 + 2 + 17 vs. Lowered Day 2 + 17
transcripts_to_geneIDs(deseq_filepath = "../graphs/DESeq2_output/hemat_transcriptomev1.6/low0_amb0217_vs_low217_indiv/AllGenes_wcols.txt", 
                       blast_filepath = "../output/BLASTs/uniprot_swissprot/hemat1.6_blastxres.tab",
                       output_path = "../output/accession_n_GOids/hemat_transcriptomev1.6/allgenes_IDs/low0_amb0217_vs_low217_indiv_All_GeneIDs.txt")

# Lowered Day 2 + 17 vs. Ambient Day 2+17
transcripts_to_geneIDs(deseq_filepath = "../graphs/DESeq2_output/hemat_transcriptomev1.6/low217_vs_amb217_indiv/AllGenes_wcols.txt", 
                       blast_filepath = "../output/BLASTs/uniprot_swissprot/hemat1.6_blastxres.tab",
                       output_path = "../output/accession_n_GOids/hemat_transcriptomev1.6/allgenes_IDs/low217_vs_amb217_indiv_All_GeneIDs.txt")

# Lowered Day 0 vs. Ambient Day 0
transcripts_to_geneIDs(deseq_filepath = "../graphs/DESeq2_output/hemat_transcriptomev1.6/low0_vs_amb0_indiv/AllGenes_wcols.txt", 
                       blast_filepath = "../output/BLASTs/uniprot_swissprot/hemat1.6_blastxres.tab",
                       output_path = "../output/accession_n_GOids/hemat_transcriptomev1.6/allgenes_IDs/low0_vs_amb0_indiv_All_GeneIDs.txt")

# All Genes
transcripts_to_geneIDs(deseq_filepath = "../graphs/DESeq2_output/hemat_transcriptomev1.6/all_indiv_libraries/crab_as_var/AllGenes_wcols.txt", 
                       blast_filepath =  "../output/BLASTs/uniprot_swissprot/hemat1.6_blastxres.tab",
                       output_path = "../output/accession_n_GOids/hemat_transcriptomev1.6/allgenes_IDs/all_indiv_libraries_All_GeneIDs.txt")
```


