---
title: "51_DEGpattern_test"
author: "Aspen Coyle"
date: "Last compiled on `r format(Sys.time(), '%Y-%m-%d')`"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

Aspen Coyle, afcoyle@uw.edu

Originally written 2021/03/10

Roberts Lab, UW-SAFS

## This is an initial attempt to use the DEGreport package - specifically the degPatterns() function - on DESeq2 output

Script is based on [this introduction](https://hbctraining.github.io/DGE_workshop/lessons/08_DGE_LRT.html)

```{r libraries, message = FALSE, warning=FALSE}
# Add all required libraries that are installed with install.packages() here
list.of.packages <- "tidyverse"
# Add all libraries that are installed using BiocManager here
bioconductor.packages <- c("DESeq2", "DEGreport")

# Install BiocManager if needed
if(!requireNamespace("BiocManager", quietly = TRUE)) install.packages("BiocManager")

# Get names of all required packages that aren't installed
new.packages <- list.of.packages[!(list.of.packages %in% installed.packages()[, "Package"])]
new.bioc.packages <- bioconductor.packages[!(bioconductor.packages %in% installed.packages()[, "Package"])]
# Install all new packages
if(length(new.packages)) install.packages(new.packages)
if(length(new.bioc.packages)) BiocManager::install(new.bioc.packages)

# Load all required libraries
all.packages <- c(list.of.packages, bioconductor.packages)
lapply(all.packages, FUN = function(X) {
  do.call("require", list(X))
})
```

### DESeq Setup

First, we'll turn the kallisto matrices from earlier into something ready for DESeq2 analysis. For this, we'll examine cbai_transcriptomev4.0, both elev day 0 vs. elev day 2 and amb day 2 vs elev day 2. Both these kallisto matrices were created in scripts 31 and 41

Our first analysis will examine:

## Bairdi Expression in Elevated Crab over Days 0 and 2



```{r deseq_variables}
# Path to kallisto matrix file
kallisto_path <- "../output/kallisto_matrices/cbai_transcriptomev4.0/elev0_vs_elev2_indiv/kallisto.isoform.counts.matrix"

# Table containing treatment information. Order should match columns of kallisto matrix file
exp_table <- data.frame(temp = factor(c("amb", "amb", "amb",
                                          "elev", "elev", "elev")), 
                               crab = factor(c("G", "H", "I", 
                                               "G", "H", "I")))
# Path to output folder
output_path <- "../graphs/DEGPatterns/cbai_transcriptomev4.0/elev0_vs_elev2_indiv"

# Variable of interest. Should match one of the factors in exp_table
variable <- "temp"
```

Now that variables are set, we should be able to easily modify future inputs. We will continue on to turning our kallisto matrix into something DESeq2 will read

``` {r deseq_formatting}
data <- read.table(kallisto_path, header = TRUE,
                     sep = "\t")
# Rename first column
names(data)[1] <- "target_ID"
  
# Set row names equal to the first column
rownames(data) <- data$target_ID
  
# Remove the now-irrelevant first column
data <- data[,-1]

# Make sure everything looks okay
print(head(data))
print(str(data))

# Round counts to integers - needed for DESeqDataSetFromMatrix()
data <- round(data, digits = 0)

# Rename rows to correspond to library numbers
rownames(exp_table) <- colnames(data)
  
# Check that experiment_table appears to match columns with matrix from Trinity
print(exp_table)
  
# Create DESeq object that looks at effect of variable
deseq2.dds <- DESeqDataSetFromMatrix(countData = (data),
                                       colData = exp_table,
                                       design = as.formula(paste0("~", variable)))

```

## Running DESeq

Unlike in the walkthrough, where the likelihood ratio test is used, we will use the default Wald test, as we just have a pairwise comparison

```{r running_DESeq}
# Perform test
dds_wald <- DESeq(deseq2.dds)
# Extract results
res_wald <- results(dds_wald)

# Subset results to return genes with padj < 0.05
sig_res_wald <- res_wald %>%
  data.frame() %>%
  rownames_to_column(var="gene") %>%
  as_tibble() %>%
  filter(padj < 0.05)

# See how many we have
nrow(sig_res_wald)

# Transform counts for data visualization
rld <- rlog(dds_wald, blind = TRUE)

# Extract rlog matrix from the object
rld_mat <- assay(rld)

# Obtain rlog values for those significant genes
res_rlog <- rld_mat[sig_res_wald$gene, ]
```

## DEGreport

Now we'll use the DEGreport package to determine sets of genes with similar expression patterns


```{r degreport}

clusters <- degPatterns(res_rlog, metadata = exp_table, time="temp", col = "crab")

plot(degPlotCluster(clusters$normalized, "temp", "crab"))

```

Now we will continue with tracking our next set of crab

### Bairdi Expression in Ambient Crab over Days 0, 2, and 17

```{r deseq_variables2}
# Path to kallisto matrix file
kallisto_path <- "../output/kallisto_matrices/cbai_transcriptomev4.0/amb0_amb2_amb17_indiv/kallisto.isoform.counts.matrix"

# Table containing treatment information. Order should match columns of kallisto matrix file
exp_table <- data.frame(day = factor(c("0", "0", "0",
                                          "2", "2", "2", "17", "17", "17")), 
                               crab = factor(c("A", "B", "C", 
                                               "A", "B", "C", "A", "B", "C")))
# Path to output folder
output_path <- "../graphs/DEGPatterns/cbai_transcriptomev4.0/amb0_amb2_amb17_indiv"

# Variable of interest. Should match one of the factors in exp_table
variable <- "day+crab"
```

Now that variables are set, we should be able to easily modify future inputs. We will continue on to turning our kallisto matrix into something DESeq2 will read

``` {r deseq_formatting2}
data <- read.table(kallisto_path, header = TRUE,
                     sep = "\t")
# Rename first column
names(data)[1] <- "target_ID"
  
# Set row names equal to the first column
rownames(data) <- data$target_ID
  
# Remove the now-irrelevant first column
data <- data[,-1]

# Make sure everything looks okay
print(head(data))
print(str(data))

# Round counts to integers - needed for DESeqDataSetFromMatrix()
data <- round(data, digits = 0)

# Rename rows to correspond to library numbers
rownames(exp_table) <- colnames(data)
  
# Check that experiment_table appears to match columns with matrix from Trinity
print(exp_table)
  
# Create DESeq object that looks at effect of variable
deseq2.dds <- DESeqDataSetFromMatrix(countData = (data),
                                       colData = exp_table,
                                       design = as.formula(paste0("~", variable)))

```

## Running DESeq

Unlike in the test above, where we used the Wald test, we'll use the LRT test since we have three comparisons

```{r running_DESeq2}
# Perform test
dds_lrt <- DESeq(deseq2.dds, test="LRT", reduced = ~crab)
# Extract results
res_lrt <- results(dds_lrt)

# Subset results to return genes with padj < 0.05
sig_res_lrt <- res_lrt %>%
  data.frame() %>%
  rownames_to_column(var="gene") %>%
  as_tibble() %>%
  filter(pvalue < 0.05)

# See how many we have
nrow(sig_res_lrt)

# Transform counts for data visualization
rld <- rlog(dds_lrt, blind = TRUE)

# Extract rlog matrix from the object
rld_mat <- assay(rld)

# Obtain rlog values for those significant genes
res_rlog <- rld_mat[sig_res_lrt$gene, ]
```

## DEGreport

Now we'll use the DEGreport package to determine sets of genes with similar expression patterns


```{r degreport2}

clusters <- degPatterns(res_rlog, metadata = exp_table, time="day", col = "crab")

plot(degPlotCluster(clusters$normalized, "day", "crab"))

```


Now, time for our third analysis - we're switching over from C. bairdi expression to Hematodinium expression

## Hemat Expression in Elevated Crab over Days 0 and 2

IMPORTANT: Rather than using only DEGs, we'll examine all genes with non-NA padj values, as we don't have enough sequences with a padj < 0.05 to graph

```{r deseq_variables3}
# Path to kallisto matrix file
kallisto_path <- "../output/kallisto_matrices/hemat_transcriptomev1.6/elev0_vs_elev2_indiv/kallisto.isoform.counts.matrix"

# Table containing treatment information. Order should match columns of kallisto matrix file
exp_table <- data.frame(temp = factor(c("amb", "amb", "amb",
                                          "elev", "elev", "elev")), 
                               crab = factor(c("G", "H", "I", 
                                               "G", "H", "I")))
# Path to output folder
output_path <- "../graphs/DEGPatterns/hemat_transcriptomev1.6/elev0_vs_elev2_indiv"

# Variable of interest. Should match one of the factors in exp_table
variable <- "temp"
```

Now that variables are set, we should be able to easily modify future inputs. We will continue on to turning our kallisto matrix into something DESeq2 will read

``` {r deseq_formatting3}
data <- read.table(kallisto_path, header = TRUE,
                     sep = "\t")
# Rename first column
names(data)[1] <- "target_ID"
  
# Set row names equal to the first column
rownames(data) <- data$target_ID
  
# Remove the now-irrelevant first column
data <- data[,-1]

# Make sure everything looks okay
print(head(data))
print(str(data))

# Round counts to integers - needed for DESeqDataSetFromMatrix()
data <- round(data, digits = 0)

# Rename rows to correspond to library numbers
rownames(exp_table) <- colnames(data)
  
# Check that experiment_table appears to match columns with matrix from Trinity
print(exp_table)
  
# Create DESeq object that looks at effect of variable
deseq2.dds <- DESeqDataSetFromMatrix(countData = (data),
                                       colData = exp_table,
                                       design = as.formula(paste0("~", variable)))

```

## Running DESeq

Unlike in the walkthrough, where the likelihood ratio test is used, we will use the default Wald test, as we just have a pairwise comparison

```{r running_DESeq3}
# Perform test
dds_wald <- DESeq(deseq2.dds)
# Extract results
res_wald <- results(dds_wald)

# Turn results to table, removing all NA values
sig_res_wald <- res_wald %>%
  data.frame() %>%
  rownames_to_column(var="gene") %>%
  as_tibble() %>%
  filter(padj < 1.01)

# See how many we have
nrow(sig_res_wald)

# Transform counts for data visualization
rld <- varianceStabilizingTransformation(dds_wald, blind = TRUE)

# Extract rlog matrix from the object
rld_mat <- assay(rld)

# Obtain rlog values for those significant genes
res_rlog <- rld_mat[sig_res_wald$gene, ]
```

## DEGreport

Now we'll use the DEGreport package to determine sets of genes with similar expression patterns


```{r degreport3}

clusters <- degPatterns(res_rlog, metadata = exp_table, time="temp", col = "crab")

plot(degPlotCluster(clusters$normalized, "temp", "crab"))

```

Now we will continue with our final analysis

### Hemat Expression in Ambient Crab over Days 0, 2, and 17

IMPORTANT: Rather than using only DEGs, we'll examine all genes with non-NA padj values, as not enough sequences have a padj < 0.05

```{r deseq_variables2}
# Path to kallisto matrix file
kallisto_path <- "../output/kallisto_matrices/hemat_transcriptomev1.6/amb0_amb2_amb17_indiv/kallisto.isoform.counts.matrix"

# Table containing treatment information. Order should match columns of kallisto matrix file
exp_table <- data.frame(day = factor(c("Day_0", "Day_0", "Day_0",
                                          "Day_2", "Day_2", "Day_2", "Day_17", "Day_17", "Day_17")), 
                               crab = factor(c("A", "B", "C", 
                                               "A", "B", "C", "A", "B", "C")))
# Path to output folder
output_path <- "../graphs/DEGPatterns/cbai_transcriptomev4.0/amb0_amb2_amb17_indiv"

# Variable of interest. Should match one of the factors in exp_table
variable <- "day+crab"
```

Now that variables are set, we should be able to easily modify future inputs. We will continue on to turning our kallisto matrix into something DESeq2 will read

``` {r deseq_formatting2}
data <- read.table(kallisto_path, header = TRUE,
                     sep = "\t")
# Rename first column
names(data)[1] <- "target_ID"
  
# Set row names equal to the first column
rownames(data) <- data$target_ID
  
# Remove the now-irrelevant first column
data <- data[,-1]

# Make sure everything looks okay
print(head(data))
print(str(data))

# Round counts to integers - needed for DESeqDataSetFromMatrix()
data <- round(data, digits = 0)

# Rename rows to correspond to library numbers
rownames(exp_table) <- colnames(data)
  
# Check that experiment_table appears to match columns with matrix from Trinity
print(exp_table)
  
# Create DESeq object that looks at effect of variable
deseq2.dds <- DESeqDataSetFromMatrix(countData = (data),
                                       colData = exp_table,
                                       design = as.formula(paste0("~", variable)))

```

## Running DESeq

Unlike in the test above, where we used the Wald test, we'll use the LRT test since we have three comparisons

```{r running_DESeq2}
# Perform test
dds_lrt <- DESeq(deseq2.dds, test="LRT", reduced = ~crab)
# Extract results
res_lrt <- results(dds_lrt)

# Subset results to return genes with padj < 0.05
sig_res_lrt <- res_lrt %>%
  data.frame() %>%
  rownames_to_column(var="gene") %>%
  as_tibble() %>%
  filter(padj < 1.01)

# See how many we have
nrow(sig_res_lrt)

# Transform counts for data visualization
rld <- rlog(dds_lrt, blind = TRUE)

# Extract rlog matrix from the object
rld_mat <- assay(rld)

# Obtain rlog values for those significant genes
res_rlog <- rld_mat[sig_res_lrt$gene, ]
```

## DEGreport

Now we'll use the DEGreport package to determine sets of genes with similar expression patterns


```{r degreport2}

clusters <- degPatterns(res_rlog, metadata = exp_table, time="day", col = "crab")

plot(degPlotCluster(clusters$normalized, "day", "crab"))

```