---
title: "Differential Expression in _Hematodinium sp._"
author: "Aidan Coyle"
date: "2/17/2021"
output: pdf_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r add_packages}
library("kableExtra")
library("tidyverse")
library("knitr")
library("magick")
```




# Methods

### Collection

Between October TK and TK, 2017, TK male _C. bairdi_ were collected from Stephen's Passage in southeastern Alaska by the Alaska Department of Fish and Game (ADF&G). This site was selected due to its reliably high rate of _Hematodinium sp._ infection - approximately 50% (ADF&G, unpub. data). Crabs were caught using pots (TK: was this part of an ADFG survey?). Once collected, crabs were transported to the (TK: NOAA facility) within TK hours, and placed in flow-through seawater tanks (TK: assuming that's what was used) at 7.5°C - the benthic water temperature within Stephen's Passage at time of capture. 

### Verification of Infection

To verify crabs were infected by _Hematodinium sp._, a TK ml aliquot of hemolymph was drawn from each crab and preserved in 800 µl 95% ethanol. 200 µl of that sample was centrifuged and pelleted. The pellet was then air-dried, and DNA was extracted with invertebrate lysis buffer. Extraction protocol followed a modified version of Ivanova et al.(2006), in which two washes with Wash Buffer were performed and eluted DNA (50 µl) was adjusted to 10 mM Tric-Cl, pH 8.0, and 0.1 mM EDTA. Once extracted, DNA was subjected to two rounds of cPCR with two primer pairs - Univ-F-15 / Hemat-R-1654 (Gruebl et al. 2002), and Hemat 18Sf / Hemat 18Sr (Bower et al. 2004). Both primer pairs are designed for the _Hematodinium spp._ small subunit (SSU) rRNA gene. After PCR, reaction aliquots were pooled and visualized on ethidium bromide-stained 2% agarose gels. If both _Hematodinium spp._ bands were visible on the gel, samples were scored as positive. If neither band was visible, samples were scored as negative. If one band amplified, samples were scored as indeterminate and (TK not selected for further analysis?)

### Temperature Variation

All crabs were acclimated to 7.5°C for nine days. 60 (TK) crabs were then selected based off cPCR results for temperature treatments and apparent recovery from capture stress. 0.2 ml of hemolymph was drawn from each selected crab and preserved in 1200 µl RNAlater. Crabs were divided randomly among six replicate tanks, with 10 crabs per tank. In three of the tanks, the water temperature was gradually raised to 10°C over a two-day period. In the three other tanks, water temperature was held at 7.5°C. At the end of the two-day acclimation period, another 0.2 ml hemolymph sample was taken from each crab and preserved in RNAlater. Following the conclusion of the experiment - 17 days after the acclimation period began - 3 samples of hemolymph were taken from each surviving crab in the ambient temperature treatment group, and 6 samples were taken from each surviving crab in the elevated temperature treatment group. The increased number of per-crab hemolymph samples in the elevated temperature treatment group was due to a mass mortality event between days 2 and 17 within that treatment group.

TK: edit methods to specify examining elev vs amb

### RNA Extraction and Sequencing

TK hemolymph samples were centrifuged at 14000 g for 10 minutes. RNA was then extracted from the pellet (TK: did it pellet?) with Quick DNA/RNA Microprep Plus Kit (Zymo Research) using the manufacturer's protocol. To quantify RNA, 2 µl samples were then run on Qubit 3.0 with the Qubit RNA HS Kit (Invitrogen). Based on RNA yield and infection severity as determined by qPCR, six crabs were selected - three from the elevated-temperature treatment group and three from the ambient-temperature treatment group. All samples from these six crabs (Table 1) were sent to the Northwest Genomics Center at Foege Hall at the University of Washington for library construction and sequencing. Due to the mass mortality event in the elevated-temperature treatment groups, no libraries from Day 17 were available for these crabs.



```{r indiv_libraries, echo = FALSE}
crabIDs <- c("A", "B", "C", "G", "H", "I")
temps <- c(rep("Ambient", 3), rep("Elevated", 3))
day0ID <- c(178, 118, 132, 173, 072, 127)
day2ID <- c(359, 349, 334, 272, 294, 280)
day17ID <- c(463, 481, 485, NA, NA, NA)

lib_table <- data.frame(crabIDs, temps, day0ID, 
                        day2ID, day17ID)
colnames(lib_table) <- c("Crab ID", "Treatment group", "Day 0 sample ID", "Day 2 sample ID", "Day 17 sample ID")

knitr::kable(lib_table, caption = "Individual libraries of infected crab", align = "c")

```


### Transcriptome Assembly and Annotation
TK. Grace's chapter describes Transcriptome 3.1. Looks similar except 3.1 has a filter - should check with Sam if Transcriptome 2.0 construction method matches. Also look at Sam's notebook posts.

### Differential Expression Analysis
An index of TK transcriptome was created with kallisto (TK citation), and each library was pseudoaligned to obtain counts. An abundance matrix for each pairwise comparison (Table 2) was then created using Trinity (v2.TK, TK citation?). Differential contig expression was calculated using a negative binomial GLM [TK: check if correct] using the R package DESeq2. Read counts were normalized using size factors and fit to a negative binomial distribution. The Wald test for significance of GLM terms for each comparison was used to obtain unadjusted p-values. For comparisons of crabs at different temperatures, a table of significantly differentially-expressed transcripts (Benjamini-Hochberg adjusted p < 0.05) was also obtained (Table 2).

```{r pairwise_comparison, echo = FALSE}

crab1ids <- c(rep("A,B,C", 4), "G,H,I")
crab1temp <- c(rep("Ambient", 5))
crab1date <- c(0, 0, 2, 2, 0)
crab2ids <- c(rep("A,B,C", 3), rep("G,H,I", 2))
crab2temp <- c(rep("Ambient", 3), rep("Elevated", 2))
crab2date <- c(2, 17, 17, 2, 2)

DEGs <- c(rep("No", 3), rep("Yes",2))

comps_table <- data.frame(crab1ids, crab1temp, crab1date,
                          crab2ids, crab2temp, crab2date,
                          DEGs)
colnames(comps_table) <- c("Crab IDs", "Temp. when sampled", "Sample day", "Crab IDs", "Temp. when sampled", "Sample day", "DEGs analyzed individually?")

kbl(comps_table, caption = "PairwiseComparisons",
    col.names = colnames(comps_table),
    align = "c") %>%
  column_spec(1:7, width = "2cm") %>%
  add_header_above(c("Pair 1" = 3, "Pair 2" = 3, " " = 1)) %>%
  kable_styling(latex_options = c("striped", "scale_down"))
  

```

### Enrichment Analysis
For each comparison (Table 2), the output from DESeq2 was cross-referenced with the annotated transcriptome and the UniProt database (citation TK) to produce a table of UniProt Accession IDs and GO terms, along with a table of UniProt Accession IDs and unadjusted p-values. GO categories were then tested for significant enrichment with the R package GO-MWU (citation TK), which utilizes the Mann-Whitney U test. 

### Individual DEG Examination
Process TK, haven't completed this yet

# Results

### DESeq2

We used the DESeq2 package to examine differential expression between libraries, and to perform various pairwise comparisons between sample groups. Principal component analyses of samples taken from the elevated-temperature treatment group showed clustering by day, and thus by temperature. This was observed for libraries aligned to both the unfiltered and host-only transcriptomes. Due to low counts, a PCA could not be created for libraries aligned to the parasite-only transcriptome. No such clustering was observed for the ambient-temperature libraries, regardless of transcriptome, along this same timeframe. [TK: CHANGE IMAGE LEGENDS FROM TEMPERATURE TO DAY].

[TK: Should we perform GO-MWU but looking at MF or CC?]

[TK: Perform GO-MWU on WGCNA module with p-val = 0.05]

[TK: RUN PCA ON ELEV 0v2 AND AMB 0v2 IMMUNE GENES?]


```{r, fig.cap = "PCA for elevated-temperature libraries, Days 0-2 (unfiltered transcriptome)", echo = FALSE}

crop <- function(im, left = 0, top = 0, right = 0, bottom = 0) {
  d <- dim(im[[1]]); w <- d[2]; h <- d[3]
  image_crop(im, glue::glue("{w-left-right}x{h-top-bottom}+{left}+{top}"))
}

"../graphs/DESeq2_output/cbai_transcriptomev2.0/elev0_vs_elev2_indiv/PCA_plot.png" %>%
  image_read() %>%
crop(top = 110, bottom = 140)

```


```{r, fig.cap = "PCA for elevated-temperature libraries, Days 0-2 (crab transcriptome)", echo = FALSE}

crop <- function(im, left = 0, top = 0, right = 0, bottom = 0) {
  d <- dim(im[[1]]); w <- d[2]; h <- d[3]
  image_crop(im, glue::glue("{w-left-right}x{h-top-bottom}+{left}+{top}"))
}

"../graphs/DESeq2_output/cbai_transcriptomev4.0/elev0_vs_elev2_indiv/PCA_plot.png" %>%
  image_read() %>%
crop(top = 110, bottom = 140)

```

```{r, fig.cap = "PCA for ambient-temperature libraries, Days 0-2 (unfiltered transcriptome)", echo = FALSE}

crop <- function(im, left = 0, top = 0, right = 0, bottom = 0) {
  d <- dim(im[[1]]); w <- d[2]; h <- d[3]
  image_crop(im, glue::glue("{w-left-right}x{h-top-bottom}+{left}+{top}"))
}

"../graphs/DESeq2_output/cbai_transcriptomev2.0/amb0_vs_amb2_indiv/PCA_plot.png" %>%
  image_read() %>%
crop(top = 110, bottom = 90)

```

```{r, fig.cap = "PCA for ambient-temperature libraries, Days 0-2 (crab transcriptome)", echo = FALSE}

crop <- function(im, left = 0, top = 0, right = 0, bottom = 0) {
  d <- dim(im[[1]]); w <- d[2]; h <- d[3]
  image_crop(im, glue::glue("{w-left-right}x{h-top-bottom}+{left}+{top}"))
}

"../graphs/DESeq2_output/cbai_transcriptomev4.0/amb0_vs_amb2_indiv/PCA_plot.png" %>%
  image_read() %>%
crop(top = 110, bottom = 90)

```

```{r, fig.cap = "PCA for ambient-temperature libraries, Days 0-2 (parasite transcriptome)", echo = FALSE}

crop <- function(im, left = 0, top = 0, right = 0, bottom = 0) {
  d <- dim(im[[1]]); w <- d[2]; h <- d[3]
  image_crop(im, glue::glue("{w-left-right}x{h-top-bottom}+{left}+{top}"))
}

"../graphs/DESeq2_output/hemat_transcriptomev1.6/amb0_vs_amb2_indiv/PCA_plot.png" %>%
  image_read() %>%
crop(top = 100, bottom = 90)

```







